react-app/src/components/CreateProposalScreen/CreateProposalScreen.tsx

```diff
@@ -1,14 +1,100 @@
-import React, { useCallback } from "react";
+import React, { useCallback, useEffect, useState } from "react";
 import cn from "classnames";
+import BigNumber from "bignumber.js";
+import { TextProposal } from "cosmjs-types/cosmos/gov/v1beta1/gov";
+import { toast } from "react-toastify";
+import { useNavigate } from "react-router-dom";
 import CreateProposalForm from "../forms/CreateProposalForm/CreateProposalForm";
 import { CreateProposalFormValues } from "../forms/CreateProposalForm/CreateProposalFormModel";
+import SubmitProposalModal from "../TransactionModals/SubmitProposalModal";
+import { SubmitProposalFormValues } from "../forms/SubmitProposalForm/SubmitProposalFormModel";
+import { useCosmos } from "../../api/cosmosAPI";
+import { useGov } from "../../api/govAPI";
+import Config from "../../config/Config";
+import { ConnectionStatus, useWallet } from "../../providers/WalletProvider";
+import { ProposalType } from "../../models/cosmos/gov";
+import { useLocale } from "../../providers/AppLocaleProvider";
+import AppRoutes from "../../navigation/AppRoutes";
 
 const CreateProposalScreen: React.FC = () => {
-  // TODO: Handle submission
-  const onSubmit = useCallback((values: CreateProposalFormValues) => {
-    console.log(values);
+  const wallet = useWallet();
+  const cosmosAPI = useCosmos();
+  const govAPI = useGov();
+  const { translate } = useLocale();
+  const navigate = useNavigate();
+  const chainInfo = Config.chainInfo;
+
+  const [isSubmissionModalActive, setIsSubmissionModalActive] =
+    useState<boolean>(false);
+  const [createProposalFormValues, setCreateProposalFormValues] =
+    useState<CreateProposalFormValues | null>(null);
+  const [userBalance, setUserBalance] = useState<BigNumber>(new BigNumber(0));
+
+  const onCloseModal = useCallback(() => {
+    setIsSubmissionModalActive(false);
   }, []);
 
+  const onSubmitCreateProposal = useCallback(
+    (values: CreateProposalFormValues) => {
+      setCreateProposalFormValues(values);
+      setIsSubmissionModalActive(true);
+    },
+    []
+  );
+
+  const onSubmitSubmitProposal = useCallback(
+    (values: SubmitProposalFormValues) => {
+      if (wallet.status !== ConnectionStatus.Connected) return;
+      const deposit = new BigNumber(values.amount).shiftedBy(
+        chainInfo.currency.coinDecimals
+      );
+
+      govAPI
```

Comment:
can we use async await here?

Reply:
in this case we might want to align all the previous calls for the shortcuts

Reply:
no time, I will open issue for the alignment